Entities/AHuman.cpp

```diff
@@ -3460,13 +3462,19 @@ void AHuman::Update()
 				heldDevice->SetAngularVel(heldDevice->GetAngularVel() + m_AngularVel * 0.5F + 3.0F * RandomNormalNum());
 
 				arm->SetHandPos(heldDevice->GetPos());
-				if (!m_Inventory.empty()) {
-					arm->SetHeldDevice(dynamic_cast<HeldDevice *>(SwapNextInventory()));
-					arm->SetHandPos(m_Pos + RotateOffset(m_HolsterOffset));
+
+                HeldDevice* BGArmDevice = m_pBGArm->GetHeldDevice();
+                if (currentlyFGArm && m_pBGArm && BGArmDevice) {
+                    m_pBGArm->RemoveAttachable(BGArmDevice, false, false);
+                    m_pFGArm->SetHeldDevice(BGArmDevice);
+                } else if (!m_Inventory.empty()) {
+                    arm->SetHeldDevice(dynamic_cast<HeldDevice*>(SwapNextInventory()));
+                    arm->SetHandPos(m_Pos + RotateOffset(m_HolsterOffset));
 				}
 				anyDropped = true;
 				break;
 			}
+            currentlyFGArm = 0;
```

Comment:
Looks like the `for` loop is set to break before this line, so it doesn't really serve its purpose.

Reply:
Loop breaks only if the foreground arm had an item and promptly dropped it, otherwise the loop then goes over the background arm, where this is needed.